### PR TITLE
SSE/InfluxDB: Change InfluxQL to work with server side expressions

### LINKED ISSUE #31604: Math expression change timestamp
**What happened**:
I first create a query and get an expected result
Screenshot 2021-03-02 at 11 50 03

Then I add an expression and timestamps is being modified on the original result as well on the calculated result
Screenshot 2021-03-02 at 11 51 25

**What you expected to happen**:
I expected the time stamp to be the same after adding the expression

**How to reproduce it (as minimally and precisely as possible)**:
Insert data to influx (i do it using node-red)
`{payload":[[{"time":1614722400000,"Price":39.729},{"Area":"SE3","uom":"öre/kWh"}],[{"time":1614718800000,"Price":38.166000000000004},{"Area":"SE3","uom":"öre/kWh"}]]}`
Create a grafana panel querying the data
i use SELECT "Price" FROM "spotprices" where time >= '2021-03-02T21:00:00.000Z'

from query inspector
```
{
  "request": {
    "method": "GET",
    "url": "api/datasources/proxy/7/query",
    "params": {
      "db": "solaredge",
      "q": "SELECT \"Price\" FROM \"spotprices\" where time >= '2021-03-02T21:00:00.000Z'",
      "epoch": "ms"
    },
    "data": null,
    "precision": "ms",
    "hideFromInspector": false
  },
  "response": {
    "results": [
      {
        "statement_id": 0,
        "series": [
          {
            "name": "spotprices",
            "columns": [
              "time",
              "Price"
            ],
            "values": [
              [
                1614718800000,
                39.729
              ],
              [
                1614722400000,
                38.166000000000004
              ]
            ]
          }
        ]
      }
    ],
    "executedQueryString": "SELECT \"Price\" FROM \"spotprices\" where time >= '2021-03-02T21:00:00.000Z'"
  }
}
```

then add a math expression $A + 15

this is the query inpspector result

```
{
  "request": {
    "url": "api/ds/query",
    "method": "POST",
    "data": {
      "queries": [
        {
          "groupBy": [
            {
              "params": [
                "$__interval"
              ],
              "type": "time"
            },
            {
              "params": [
                "none"
              ],
              "type": "fill"
            }
          ],
          "measurement": "spotprices",
          "orderByTime": "ASC",
          "policy": "default",
          "query": "SELECT \"Price\" FROM \"spotprices\" where time >= '2021-03-02T21:00:00.000Z'",
          "rawQuery": true,
          "refId": "A",
          "resultFormat": "time_series",
          "select": [
            [
              {
                "params": [
                  "Price"
                ],
                "type": "field"
              },
              {
                "params": [],
                "type": "mean"
              }
            ]
          ],
          "tags": [],
          "datasource": "solaredge",
          "datasourceId": 7,
          "intervalMs": 200,
          "maxDataPoints": 1488
        },
        {
          "refId": "B",
          "type": "math",
          "datasource": "__expr__",
          "hide": false,
          "expression": "$A+15",
          "datasourceId": -100
        }
      ],
      "range": {
        "from": "1970-01-19T16:24:54.000Z",
        "to": "1970-01-19T16:32:02.400Z",
        "raw": {
          "from": "1970-01-19T16:24:54.000Z",
          "to": "1970-01-19T16:32:02.400Z"
        }
      },
      "from": "1614294000",
      "to": "1614722400"
    },
    "hideFromInspector": false
  },
  "response": {
    "results": {
      "A": {
        "refId": "A",
        "series": null,
        "tables": null,
        "dataframes": [
          "QVJST1cxAAD/////eAEAABAAAAAAAAoADgAMAAsABAAKAAAAFAAAAAAAAAEDAAoADAAAAAgABAAKAAAACAAAAGAAAAACAAAAKAAAAAQAAAAY////CAAAAAwAAAABAAAAQQAAAAUAAAByZWZJZAAAADj///8IAAAAHAAAABAAAABzcG90cHJpY2VzLlByaWNlAAAAAAQAAABuYW1lAAAAAAIAAAB0AAAABAAAAKb///8UAAAAPAAAADwAAAAAAAMBPAAAAAEAAAAEAAAAlP///wgAAAAQAAAABQAAAHZhbHVlAAAABAAAAG5hbWUAAAAAAAAAAIr///8AAAIABQAAAHZhbHVlABIAGAAUABMAEgAMAAAACAAEABIAAAAUAAAARAAAAEwAAAAAAAoBTAAAAAEAAAAMAAAACAAMAAgABAAIAAAACAAAABAAAAAEAAAAdGltZQAAAAAEAAAAbmFtZQAAAAAAAAAAAAAGAAgABgAGAAAAAAADAAQAAAB0aW1lAAAAAAAAAAD/////uAAAABQAAAAAAAAADAAWABQAEwAMAAQADAAAACAAAAAAAAAAFAAAAAAAAAMDAAoAGAAMAAgABAAKAAAAFAAAAFgAAAACAAAAAAAAAAAAAAAEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAAQAAAAAAAAAAAAAAACAAAAAgAAAAAAAAAAAAAAAAAAAAIAAAAAAAAAAAAAAAAAAAAAdHwElLwFAAAYENuUvAUAWmQ730/dQ0Bpke18PxVDQBAAAAAMABQAEgAMAAgABAAMAAAAEAAAACwAAAA4AAAAAAADAAEAAACIAQAAAAAAAMAAAAAAAAAAIAAAAAAAAAAAAAAAAAAAAAAACgAMAAAACAAEAAoAAAAIAAAAYAAAAAIAAAAoAAAABAAAABj///8IAAAADAAAAAEAAABBAAAABQAAAHJlZklkAAAAOP///wgAAAAcAAAAEAAAAHNwb3RwcmljZXMuUHJpY2UAAAAABAAAAG5hbWUAAAAAAgAAAHQAAAAEAAAApv///xQAAAA8AAAAPAAAAAAAAwE8AAAAAQAAAAQAAACU////CAAAABAAAAAFAAAAdmFsdWUAAAAEAAAAbmFtZQAAAAAAAAAAiv///wAAAgAFAAAAdmFsdWUAEgAYABQAEwASAAwAAAAIAAQAEgAAABQAAABEAAAATAAAAAAACgFMAAAAAQAAAAwAAAAIAAwACAAEAAgAAAAIAAAAEAAAAAQAAAB0aW1lAAAAAAQAAABuYW1lAAAAAAAAAAAAAAYACAAGAAYAAAAAAAMABAAAAHRpbWUAAAAAoAEAAEFSUk9XMQ=="
        ]
      },
      "B": {
        "refId": "B",
        "series": null,
        "tables": null,
        "dataframes": [
          "QVJST1cxAAD/////YAEAABAAAAAAAAoADgAMAAsABAAKAAAAFAAAAAAAAAEDAAoADAAAAAgABAAKAAAACAAAAFAAAAACAAAAKAAAAAQAAAAw////CAAAAAwAAAABAAAAQgAAAAUAAAByZWZJZAAAAFD///8IAAAADAAAAAAAAAAAAAAABAAAAG5hbWUAAAAAAgAAAGwAAAAEAAAArv///xQAAAA4AAAAOAAAAAAAAwE4AAAAAQAAAAQAAACc////CAAAAAwAAAABAAAAQgAAAAQAAABuYW1lAAAAAAAAAACO////AAACAAEAAABCABIAGAAUABMAEgAMAAAACAAEABIAAAAUAAAARAAAAEwAAAAAAAoBTAAAAAEAAAAMAAAACAAMAAgABAAIAAAACAAAABAAAAAEAAAAVGltZQAAAAAEAAAAbmFtZQAAAAAAAAAAAAAGAAgABgAGAAAAAAADAAQAAABUaW1lAAAAAAAAAAD/////uAAAABQAAAAAAAAADAAWABQAEwAMAAQADAAAACAAAAAAAAAAFAAAAAAAAAMDAAoAGAAMAAgABAAKAAAAFAAAAFgAAAACAAAAAAAAAAAAAAAEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAAQAAAAAAAAAAAAAAACAAAAAgAAAAAAAAAAAAAAAAAAAAIAAAAAAAAAAAAAAAAAAAAAdHwElLwFAAAYENuUvAUAWmQ7309dS0Bpke18P5VKQBAAAAAMABQAEgAMAAgABAAMAAAAEAAAACwAAAA4AAAAAAADAAEAAABwAQAAAAAAAMAAAAAAAAAAIAAAAAAAAAAAAAAAAAAAAAAACgAMAAAACAAEAAoAAAAIAAAAUAAAAAIAAAAoAAAABAAAADD///8IAAAADAAAAAEAAABCAAAABQAAAHJlZklkAAAAUP///wgAAAAMAAAAAAAAAAAAAAAEAAAAbmFtZQAAAAACAAAAbAAAAAQAAACu////FAAAADgAAAA4AAAAAAADATgAAAABAAAABAAAAJz///8IAAAADAAAAAEAAABCAAAABAAAAG5hbWUAAAAAAAAAAI7///8AAAIAAQAAAEIAEgAYABQAEwASAAwAAAAIAAQAEgAAABQAAABEAAAATAAAAAAACgFMAAAAAQAAAAwAAAAIAAwACAAEAAgAAAAIAAAAEAAAAAQAAABUaW1lAAAAAAQAAABuYW1lAAAAAAAAAAAAAAYACAAGAAYAAAAAAAMABAAAAFRpbWUAAAAAiAEAAEFSUk9XMQ=="
        ]
      }
    }
  }
}
```

**Anything else we need to know?**:

**Environment**:
- Grafana version:
 * Latest docker, v7.5.0-12932pre (4c0e2c3174)
- Data source type & version:
 * InfluxDB version: 1.8.3, Fluxql
- OS Grafana is installed on:
 * ubuntu using docker
- User OS & Browser:
 * MAC os big sur
 * Chrome 88.0.4324.192
- Grafana plugins:
  * CompareQueries from AutohomeCorp
  * Discrete from Natel Energy
- Others:

### PR DESCRIPTION
**Pull request by kylebrandt**
**What this PR does / why we need it**:
Updates the request to handle EpochMS as the time range (which is what expressions sends) and changes the response to be the response to use millisecond timestamps instead of seconds.

**Which issue(s) this PR fixes**:

Fixes #31604

**Special notes for your reviewer**:
Similar to https://github.com/grafana/grafana/pull/31455

### CODE REVIEW

**Comment by kylebrandt** [comment]
Before returned "No data", now returns:

image

File: pkg/tsdb/influxdb/query.go [hunk 1]
```diff
@@ -78,7 +78,24 @@ func (query *Query) renderTags() []string {
...
+	if _, err := strconv.ParseInt(tr.To, 10, 64); err != nil {
+		return false
+	}
+	return true
+}
+
 func (query *Query) renderTimeFilter(queryContext plugins.DataQuery) string {
+	// If from expressions
+	if isTimeRangeNumeric(queryContext.TimeRange) {
+		from, to, _ := epochMStoInfluxTime(queryContext.TimeRange)
```
Comment (aknuds1) [inline pkg/tsdb/influxdb/query.go:94]: Shouldn't the error be handled?

Comment (kylebrandt) [reply pkg/tsdb/influxdb/query.go:94]: Right forgot about that. These don't return errors themselves, but I imagine I should at least log?

Comment (aknuds1) [reply pkg/tsdb/influxdb/query.go:94]: @kylebrandt whatever's appropriate, I don't know the context :)

File: pkg/tsdb/influxdb/query.go [hunk 2]
```diff
@@ -78,7 +78,29 @@ func (query *Query) renderTags() []string {
...
+	if _, err := strconv.ParseInt(tr.To, 10, 64); err != nil {
+		return false
+	}
+	return true
+}
+
 func (query *Query) renderTimeFilter(queryContext plugins.DataQuery) string {
+	// If from expressions
+	if isTimeRangeNumeric(queryContext.TimeRange) {
+		from, to, err := epochMStoInfluxTime(queryContext.TimeRange)
```
Comment (papagian) [inline pkg/tsdb/influxdb/query.go:94]: a minor remark: `isTimeRangeNumeric` and `epochMStoInfluxTime` are pretty similar so maybe the condition could change and check if the `epochMStoInfluxTime` returns an error to fall back to non-expression time range processing.

**Review by papagian — APPROVED** [review]
AFAICT it looks good and I've tested and works fine.

**Pull request merged**